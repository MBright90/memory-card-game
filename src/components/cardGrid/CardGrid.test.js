import { appContext } from '@app'
import CardGrid from '@components/app'
import { render, screen } from '@testing-library/react'
import React from 'react'

describe('CardGrid', () => {
  let characterArr

  beforeEach(() => {
    characterArr = [
      {
        character: 'Jar-Jar Binks',
        identifier: 'jarJarBinks',
        imageUrl: 'www.example.com/JarJarBinks',
        tabIndex: 0
      },
      {
        character: 'Jyn Erso',
        identifier: 'jynErso',
        imageUrl: 'www.example.com/JynErso',
        tabIndex: 0
      },
      {
        character: 'Poe Dameron',
        identifier: 'poeDameron',
        imageUrl: 'www.example.com/poeDameron',
        tabIndex: 0
      }
    ]
  })

  it('renders all card elements when passed a valid character array', () => {
    const context = { characterArr }
    render(<CardGrid />, {
      wrapper: ({ children }) => (
        <appContext.Provider value={context}>{children}</appContext.Provider>
      )
    })

    const JarJarCard = screen.getByText(/Jar-Jar Binks/i)
    expect(JarJarCard).toBeInTheDocument()

    const JynCard = screen.getByText(/Jyn Erso/i)
    expect(JynCard).toBeInTheDocument()

    const poeCard = screen.getByText(/Poe Dameron/i)
    expect(poeCard).toBeInTheDocument()
  })

  it('renders the cards image for a character passed in', () => {
    const context = { characterArr }
    render(<CardGrid />, {
      wrapper: ({ children }) => (
        <appContext.Provider value={context}>{children}</appContext.Provider>
      )
    })

    const cardImageArr = screen.getAllByRole('img')
    cardImageArr.forEach((cardImage) => {
      expect(cardImage.style.backgroundImage).toBe(/www.example.com\W/i)
    })
  })
})
