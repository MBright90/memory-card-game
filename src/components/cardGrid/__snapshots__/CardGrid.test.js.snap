// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardGrid renders to match snapshot 1`] = `
<div>
  <div
    class="cardGrid"
  >
    <div
      class="card"
      data-identifier="poeDameron"
      role="button"
      tabindex="1"
    >
      <p
        class="cardTitle"
      >
        Poe Dameron
      </p>
      <div
        class="cardImage"
        role="img"
        style="background-image: url(www.example.com/poeDameron);"
      />
    </div>
    <div
      class="card"
      data-identifier="jynErso"
      role="button"
      tabindex="2"
    >
      <p
        class="cardTitle"
      >
        Jyn Erso
      </p>
      <div
        class="cardImage"
        role="img"
        style="background-image: url(www.example.com/JynErso);"
      />
    </div>
    <div
      class="card"
      data-identifier="jarJarBinks"
      role="button"
      tabindex="3"
    >
      <p
        class="cardTitle"
      >
        Jar-Jar Binks
      </p>
      <div
        class="cardImage"
        role="img"
        style="background-image: url(www.example.com/JarJarBinks);"
      />
    </div>
  </div>
</div>
`;
